packages/docsify-server-renderer/index.js

```diff
@@ -41,21 +42,22 @@ function isExternal(url) {
 }
 
 function mainTpl(config) {
-  let html = `<nav class="app-nav${
-    config.repo ? '' : ' no-badge'
-  }"><!--navbar--></nav>`;
-
-  if (config.repo) {
-    html += tpl.corner(config.repo);
-  }
-
-  if (config.coverpage) {
-    html += tpl.cover();
-  }
-
-  html += tpl.main(config);
-
-  return html;
+  return (
+    <>
+      <nav
+        class={`app-nav${config.repo ? '' : ' no-badge'}`}
+        innerHTML={'<!--navbar-->'}
+      ></nav>
+      {config.repo && (
+        <GithubCorner
+          githubUrl={config.repo}
+          cornerExternalLinkTarget={config.cornerExternalLinkTarge}
+        />
+      )}
+      {config.coverpage && <Cover />}
+      <Main {...config} />
+    </>
+  );
```

Comment:
By converting to Solid components, like this example, we will eventually be able to allow end users to provide their own component in order to override Docsify's, allowing them to make their own Docsify templates.

This will vastly improve Docsify's theme-ability: not only can they override CSS like currently (f.e. docsify-themeable being the most notable example), but they will also be able to replace the DOM structure that wraps the markdown output.